python/foglamp/plugins/south/sinusoid/sinusoid.py

```diff
@@ -180,11 +181,13 @@ def generate_data():
                                           timestamp=data['timestamp'], key=data['key'],
                                           readings=data['readings'])
 
-                await asyncio.sleep(1 / (int(handle['dataPointsPerSec']['value'])))
-
+                try:
+                    await asyncio.sleep(1 / (float(handle['dataPointsPerSec']['value'])))
+                except ZeroDivisionError:
+                    _LOGGER.warning('Data points per second must be greater than 0, defaulting to sleep time 1')
+                    await asyncio.sleep(1)
```

Comment:
Better to set handle['dataPointsPerSec']['value'] = '1.0' to avoid overcrowding syslog with repeated log messages.